scipy/optimize/_linprog_util.py

```diff
@@ -1126,13 +1126,20 @@ def vstack(blocks):
             A_eq[:, i_nolb] *= -1
 
     # upper bound: add inequality constraint
-    i_newub = np.nonzero(ub_some)[0]
+    i_newub, = ub_some.nonzero()
     ub_newub = ubs[ub_some]
-    n_bounds = np.count_nonzero(ub_some)
-    A_ub = vstack((A_ub, zeros((n_bounds, A_ub.shape[1]))))
-    b_ub = np.concatenate((b_ub, np.zeros(n_bounds)))
-    A_ub[range(m_ub, A_ub.shape[0]), i_newub] = 1
-    b_ub[m_ub:] = ub_newub
+    n_bounds = len(i_newub)
+    if n_bounds > 0:
+        shape = (n_bounds, A_ub.shape[1])
+        if sparse:
```

Comment:
For sparse matrices, generate the new rows with ones, then append them (as opposed to appending zero rows, then filling in ones).